apps/api/src/app/environments/usecases/create-environment/create-environment.usecase.ts

```diff
@@ -1,9 +1,10 @@
+import { nanoid } from 'nanoid';
```

Comment:
could you please also cover this functionality with e2e tests for all use-cases touched?

Reply:
I was not sure what would be the best way to test it as the API key is i the code of our testing environment. i wanted to revisit the tests tomorrow once again.

small note, our test env organically simulates backward compatibility because we do not store an encrypted api key by the @novu/test service on sesstion init. i could not add the encryption because we do not propose application genetic in test lib.
meaning that we created not encrypted api keys in the DB on session init and the application (e2e tests) works as expected (backward compatibility).

Reply:
I see, we should have in the `EnvironmentService` the same encryption ideally by reusing the utils `encryptApiKey, decryptApiKey`.
Theoretically, both packages are built to CJS, so we might include `@novu/application-generic` in `@novu/testing`, but not suggesting that 😅 Another way would be to have another lib that is pure for reusable utils and without app-related logic, which is headache 🤕 so the only easy thing would be to duplicate the code?